packages/graphql-codegen-core/src/utils/parse.ts

```diff
@@ -0,0 +1,17 @@
+import { parse as coreParse, Source } from 'graphql';
+
+export function parse(source: Source | string) {
+  try {
+    return coreParse(source);
+  } catch (e) {
+    let error: Error;
+
+    if (typeof source === 'string') {
+      error = new Error(`Parsing '${source.trim().substr(0, 25)}...' failed with: ${e.message}`);
```

Comment:
I don't think caught error's stack is worth to attach to new error because it's mostly internal stack of `graphql`

Reply:
@dotansimha

Reply:
Yeah it's fine